src/Store/CookieStore.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace Auth0\SDK\Store;
+
+/**
+ * This class provides a layer to persist transient auth data using cookies.
+ */
+class CookieStore implements StoreInterface
+{
+    /**
+     * Default cookie base name.
+     */
+    const BASE_NAME = 'auth0';
+
+    /**
+     * Cookie base name, configurable on instantiation.
+     *
+     * @var string
+     */
+    protected $cookie_base_name;
+
+    /**
+     * Cookie expiration, configurable on instantiation.
+     *
+     * @var integer
+     */
+    protected $cookie_expiration;
+
+    /**
+     * CookieStore constructor.
+     *
+     * @param string  $base_name Cookie base name.
+     * @param integer $expires   Cookie expiration length, in seconds.
+     */
+    public function __construct(?string $base_name = self::BASE_NAME, ?int $expires = 600)
+    {
+        $this->cookie_base_name  = $base_name;
+        $this->cookie_expiration = $expires;
+    }
+
+    /**
+     * Persists $value on cookies, identified by $key.
+     *
+     * @param string $key   Cookie to set.
+     * @param mixed  $value Value to use.
+     *
+     * @return void
+     */
+    public function set($key, $value)
+    {
+        $key_name           = $this->getCookieName($key);
+        $_COOKIE[$key_name] = $value;
```

Comment:
what is the difference between this line and the one below?

Reply:
`$_COOKIE` is the in-memory global of all cookie values received form the browser and `setcookie()` sends the `Set-Cookie` HTTP header